repository/repository.go

```diff
@@ -113,22 +162,36 @@ type AppLimits struct {
 	NotificationSettings *NotificationSettings `json:"notificationSettings,omitempty"`
 }
 
-type PayPlan struct {
-	PlanType   PayPlanType `json:"planType"`
-	DailyLimit int         `json:"dailyLimit"`
-}
-
 // UpdateApplication struct holding possible fields to update
 type UpdateApplication struct {
 	Name                 string                `json:"name,omitempty"`
 	Status               AppStatus             `json:"status,omitempty"`
-	PayPlanType          PayPlanType           `json:"payPlanType,omitempty"`
 	FirstDateSurpassed   time.Time             `json:"firstDateSurpassed,omitempty"`
 	GatewaySettings      *GatewaySettings      `json:"gatewaySettings,omitempty"`
 	NotificationSettings *NotificationSettings `json:"notificationSettings,omitempty"`
+	Limit                *AppLimit             `json:"appLimit,omitempty"`
 	Remove               bool                  `json:"remove,omitempty"`
 }
 
+func (u *UpdateApplication) IsInvalid() error {
```

Comment:
Same comment as before regarding the naming, but still not a blocker.

Reply:
agreed, Validate is a better naming, I'll update before merging.